spec/lib/api/2.0/skus-spec.js

```diff
@@ -3,234 +3,238 @@
+            nodeApiService = helper.injector.get('Http.Services.Api.Nodes');
+            sinon.stub(nodeApiService);
+            skuApiService = helper.injector.get('Http.Services.SkuPack');
+            Promise = helper.injector.get('Promise');
+            Constants = helper.injector.get('Constants');
+            Errors = helper.injector.get('Errors');
         });
+
     });
 
-    beforeEach('reset test DB', function () {
-        return helper.reset();
+    beforeEach('reset stubs', function () {
+        function resetStubs(obj) {
+            _(obj).methods().forEach(function (method) {
+                if (obj[method] && obj[method].reset) {
+                    obj[method].reset();
+                }
+            }).value();
+        }
+        resetStubs(waterline.nodes);
+        resetStubs(waterline.skus);
+        resetStubs(workflowApiService);
+
     });
 
     after('stop HTTP server', function () {
         return helper.stopServer();
     });
 
+    var node = {
+        id: '1234abcd1234abcd1234abcd',
+        name: 'sku test node',
+        type: 'compute',
+        obmSettings: [
+            {
+                service: 'ipmi-obm-service',
+                config: {
+                    host: '1.2.3.4',
+                    user: 'myuser',
+                    password: 'mypass'
+                }
+            }
+        ],
+        autoDiscover: "false",
+        identifiers: [],
+        createdAt: '010101',
+        updatedAt: '010101',
+        tags: [],
+        sku: '0987'
+    };
+
+    var input = {
+        name: 'my test sku',
+        rules: [
+            {
+                path: 'dmi.dmi.base_board.manufacturer',
+                contains: 'Intel'
+            },
+            {
+                path: 'dmi.memory.total',
+                equals: '32946864kB'
+            }
+        ],
+        discoveryGraphName: 'TestGraph.Dummy',
+        discoveryGraphOptions: { test: 1 },
+        id: '0987'
+    };
+
+    beforeEach(function() {
+        sinon.stub(skuApiService, 'getSkus');
+        sinon.stub(skuApiService, 'postSku');
+        sinon.stub(skuApiService, 'getSkusById');
+        sinon.stub(skuApiService, 'upsertSku');
+        sinon.stub(skuApiService, 'patchSku');
+        sinon.stub(skuApiService, 'getNodesSkusById');
+        sinon.stub(skuApiService, 'regenerateSkus');
+        sinon.stub(skuApiService, 'deleteSkuById');
+        sinon.stub(workflowApiService, 'createAndRunGraph');
+    });
+
+    afterEach(function() {
+        skuApiService.getSkus.restore();
+        skuApiService.postSku.restore();
+        skuApiService.getSkusById.restore();
+        skuApiService.upsertSku.restore();
+        skuApiService.patchSku.restore();
+        skuApiService.regenerateSkus.restore();
+        skuApiService.getNodesSkusById.restore();
+        skuApiService.deleteSkuById.restore();
+        workflowApiService.createAndRunGraph.restore();
```

Comment:
A recommendation: Use  the `sinon.sandbox` will save the effort to restore these stubs one by one.